HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/ShareInvoiceBottomView.swift

```diff
@@ -0,0 +1,411 @@
+            continueButton.topAnchor.constraint(equalTo: continueView.topAnchor, constant: Constants.topBottomPaddingConstraint),
+            continueButton.bottomAnchor.constraint(equalTo: continueView.bottomAnchor)
+        ])
+    }
+
+    private func setupPoweredByGiniConstraints() {
+        NSLayoutConstraint.activate([
+            bottomStackView.leadingAnchor.constraint(equalTo: bottomView.leadingAnchor, constant: Constants.viewPaddingConstraint),
+            bottomStackView.trailingAnchor.constraint(equalTo: bottomView.trailingAnchor, constant: -Constants.viewPaddingConstraint),
+            bottomStackView.topAnchor.constraint(equalTo: bottomView.topAnchor, constant: Constants.topAnchorPoweredByGiniConstraint),
+            bottomStackView.bottomAnchor.constraint(equalTo: bottomView.bottomAnchor)
+        ])
+    }
+    
+    @objc
+    private func tapOnContinueButton() {
+        viewModel.didTapOnContinue()
+    }
+
+    @objc
+    private func tapOnAppStoreButton() {
+        openPaymentProvidersAppStoreLink(urlString: viewModel.selectedPaymentProvider?.appStoreUrlIOS)
+    }
+    
+    @objc
+    private func tapOnLabelAction(gesture: UITapGestureRecognizer) {
+        if gesture.didTapAttributedTextInLabel(label: tipLabel,
+                                               targetText: viewModel.tipActionablePartText) {
+            tapOnAppStoreButton()
+        }
+    }
+    
+    private func openPaymentProvidersAppStoreLink(urlString: String?) {
+        guard let urlString = urlString else {
+            print("AppStore link unavailable for this payment provider")
+            return
+        }
+        if let url = URL(string: urlString), UIApplication.shared.canOpenURL(url) {
+            UIApplication.shared.open(url)
+        }
+    }
+    
+    private func generateAppViews() -> [SingleAppView] {
+        var viewsToReturn: [SingleAppView] = []
+        viewModel.appsMocked.forEach { singleApp in
+            let view = SingleAppView()
+            view.configure(image: singleApp.image, title: singleApp.title)
+            viewsToReturn.append(view)
+        }
+        return viewsToReturn
+    }
+}
+
+extension ShareInvoiceBottomView {
+    enum Constants {
+        static let viewPaddingConstraint = 16.0
+        static let topBottomPaddingConstraint = 10.0
+        static let bottomDescriptionConstraint = 20.0
+        static let bankIconSize = 36
+        static let bankIconCornerRadius = 6.0
+        static let bankIconBorderWidth = 1.0
+        static let continueButtonViewHeight = 56.0
+        static let appsViewSpacing: CGFloat = 4.0
+        static let appsViewHeight: CGFloat = 112.0
+        static let topAnchorAppsViewConstraint = 20.0
+        static let trailingAppsViewConstraint = 50.0
+        static let topAnchorTipViewConstraint = 5.0
+        static let topAnchorPoweredByGiniConstraint = 5.0
+        static let tipIconSize = 24.0
+    }
+}
+
+class SingleAppView: UIView {
```

Comment:
[Suggestion]: Could you please extract `SingleAppView` class to a dedicated file?